src/__test__/components/plots-and-tables/markerHeatmap.test.jsx

```diff
@@ -0,0 +1,268 @@
+
+  rtl.render(
+    <Provider store={store}>
+      <MarkerHeatmap
+        experimentId={experimentId}
+      />
+    </Provider>,
+  );
+  await rtl.waitFor(() => expect(loadConfigSpy).toHaveBeenCalledTimes(1));
+};
+
+describe('Marker heatmap plot', () => {
+  beforeAll(async () => {
+    await preloadAll();
+  });
+  beforeEach(() => {
+    jest.clearAllMocks();
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    fetchMock.mockResponse(JSON.stringify({}), { status: 404, statusText: '404 Not Found' });
+    loadConfigSpy = jest.spyOn(loadConfig, 'default');
+    loadMarkersSpy = jest.spyOn(markerGenesLoaded, 'default');
+    configUpdatedSpy = jest.spyOn(configUpdated, 'default');
+    cellSetsLoadedSpy = jest.spyOn(cellSetsLoaded, 'default');
+    loadedProcessingConfigSpy = jest.spyOn(loadedProcessingConfig, 'default');
+  });
+
+  it('loads initially', async () => {
+    await renderHeatmapPage(defaultStore);
+    expect(loadMarkersSpy).toHaveBeenCalled();
+    expect(configUpdatedSpy).toHaveBeenCalled();
+  });
+
+  it('loads marker genes on selecting', async () => {
+    await renderHeatmapPage(defaultStore);
+
+    const geneSelection = rtl.screen.getByText('Gene selection');
+    userEvent.click(geneSelection);
+    const nGenesInput = rtl.getByTestId(geneSelection.parentElement, 'num-genes');
+    userEvent.type(nGenesInput, 10);
+
+    const runButton = rtl.getByText(geneSelection.parentElement, 'Run');
+    userEvent.click(runButton);
+
+    await rtl.waitFor(() => expect(loadMarkersSpy).toHaveBeenCalled());
+  });
+
+  it('sorts genes properly when adding a gene', async () => {
+    await renderHeatmapPage(defaultStore);
+    await rtl.waitFor(() => expect(configUpdatedSpy).toHaveBeenCalled());
+    store.dispatch(loadGeneExpression(experimentId, ['gene0', 'gene1', 'gene3', 'gene2'], plotUuid));
+    await rtl.waitFor(() => expect(configUpdatedSpy).toHaveBeenCalledTimes(2));
+    expect(store.getState().componentConfig[plotUuid].config.selectedGenes).toEqual(['gene0', 'gene1', 'gene2', 'gene3']);
+    expect(loadMarkersSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('removing a gene keeps the sorted order without re-sorting', async () => {
+    await renderHeatmapPage(defaultStore);
+    await rtl.waitFor(() => expect(configUpdatedSpy).toHaveBeenCalled());
+    store.dispatch(loadGeneExpression(experimentId, ['gene0', 'gene3'], plotUuid));
+    await rtl.waitFor(() => expect(configUpdatedSpy).toHaveBeenCalledTimes(3));
+    expect(store.getState().componentConfig[plotUuid].config.selectedGenes).toEqual(['gene0', 'gene3']);
+    expect(loadMarkersSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('loads cellsets if not available', async () => {
+    const newStore = { ...defaultStore, cellSets: { loading: true } };
+    await renderHeatmapPage(newStore);
+    await rtl.waitFor(() => expect(cellSetsLoadedSpy).toHaveBeenCalled());
+  });
+
+  it('loads processing settings if louvainresolution is not available', async () => {
```

Comment:
nice, this is what I meant on the call. Also add a test with `newStore` equal to what is its initial value, then try to load the marker heatmap and assert that the heatmap goes into loading state (Skeleton is displayed) and that dispatches for the relevant actions have been triggered.